src/geometry/euclidean/oriented_angle.lean

```diff
@@ -773,3 +780,424 @@ begin
 end
 
 end orthonormal
+
+namespace orientation
+
+open finite_dimensional
+
+variables {V : Type*} [inner_product_space ℝ V]
+variables [hd2 : fact (finrank ℝ V = 2)] (o : orientation ℝ V (fin 2))
+include hd2 o
+
+local notation `ob` := o.fin_orthonormal_basis_orthonormal dec_trivial hd2.out
+
+/-- The oriented angle from `x` to `y`, modulo `2 * π`. If either vector is 0, this is 0. -/
+def oangle (x y : V) : real.angle :=
```

Comment:
Do we have a plan for when we want to talk about the angle between two vectors which live in a higher-dimensional space?

We could of course factor through this definition by defining such an angle to be this definition applied to the vectors regarded as members of their span but this might be awkward, especially if they are linearly dependent.

Before I saw this proposal I would have expected us to define the angle between two vectors using the `d`-dimensional cosine rule (supported by Cauchy-Schwarz).

Reply:
We already have *unoriented* angles in spaces of any dimension, which is what you get from the cosine rule. See `geometry.euclidean.basic` for `inner_product_geometry.angle` and `euclidean_geometry.angle`. *Oriented* angles between two vectors only make sense when you have an orientation of a two-dimensional space containing them. (The relation between oriented and unoriented angles is intended for further followups. If you want to use oriented angles to prove something about unoriented angles in a higher-dimensional space, I think moving between the space and a subspace to use oriented angles is probably appropriate, though I expect we'll need more lemmas in due course to transport geometrical results across isometries like that.)

A case of oriented angles that makes more sense in higher dimensions is that of the oriented angle between two oriented (n+1)-dimensional subspaces in an (n+2)-dimensional space, given also an orientation of a two-dimensional subspace orthogonal to an n-dimensional subspace contained in both (n+1)-dimensional subspaces (in two dimensions, that reduces to the oriented angle between two vectors). That's relevant for talking about angle bisectors for the angles between such (n+1)-dimensional subspaces, which is relevant for stating that the incenter of a simplex is the intersection of the angle bisectors, for example. But that sort of thing, involving orientations of three subspaces simultaneously, is probably some way away.

Reply:
I see, thank you for the detailed response!

Reply:
I think it would be good to add a summary of these words:
> We already have unoriented angles in spaces of any dimension, which is what you get from the cosine rule. See geometry.euclidean.basic for inner_product_geometry.angle and euclidean_geometry.angle.

to the doc string here.

Reply:
I've added cross references in both directions between `orientation.oangle` and `inner_product_geometry.angle`.